### ISSUE TITLE
gradle init run in the root of filesystem fails with a non actionnable error message

### ISSUE THREAD

**Issue by biergit** [gradle/gradle#22620]
Hi,
I used Gradle 7.5.1 on Ubuntu 22.04 and issued `gradle init` and then gradle said goodbye but left an error message:
```
FAILURE: Build failed with an exception.

* Where:
Settings file '/settings.gradle' line: 18

* What went wrong:
A problem occurred evaluating settings ''.
The project name must not be empty. Set the 'rootProject.name' or adjust the 'include' statement (see https://docs.gradle.org/7.5.1/dsl/org.gradle.api.initialization.Settings.html#org.gradle.api.initialization.Settings:include(java.lang.String[]) for more details).

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 619ms
```
This is in constrast to what is promised in the documentation e.g. [here](https://docs.gradle.org/current/samples/sample_building_java_libraries.html)

**Comment by eskatos** [comment]
Thank you for providing a valid reproducer.

The issue is in the backlog of the relevant team but the existence of a workaround makes it non-critical so it might take a while before a fix is made.

----

It appears that you are running `gradle init` in the root of your filesystem `/`.
This is not supported but the error message could be better.
I edited the title of the issue in this regard.
As a workaround please run `gradle init` inside a folder.

**Comment by eskatos** [comment]
This issue is a good choice for first-time contributors to Gradle, it is actionable and ready for contribution.

See [CONTRIBUTING.md](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md) for more information.

**Comment by biergit** [comment]
@eskatos I was not issuing the command at the root of the filesystem:

![gradle-init-fail](https://user-images.githubusercontent.com/1071968/202704640-74a90fe9-23ce-4592-812b-9ad2f2a80369.png)

**Comment by eskatos** [comment]
In the error message the path to the settings script is `/settings.gradle` which is at the root of the filesystem.

Could it be that you don't have a settings script in your project folder and that there is one at the root of the filesystem?

**Comment by biergit** [comment]
The project folder is empty and the root also contains no  `settings.gradle` file.
FYI I used a plain Ubuntu container and installed Gradle via SDKMAN!.

**Comment by eskatos** [comment]
I tried to reproduce in a vscode ubuntu dev-container and I could not:

![image](https://user-images.githubusercontent.com/132773/203999243-502dc5c8-a236-4b8f-bdf4-bd3c6865f6ec.png)

----

What the error message shown in your screenshots says is that Gradle found a `settings.gradle` script which does not set the `rootProject.name`. In that case Gradle tries to derive the rootProject.name from the name of the folder containing the `settings.gradle`. And that fails if the `settings.gradle` sits at the root. I was able to reproduce by putting a `settings.gradle` file in the root of my filesystem.

Any idea how you could help me reproduce?

**Comment by biergit** [comment]
Hi @eskatos , I tried again and also failed to reproduce it, sorry.

**Comment by eskatos** [comment]
Thanks for trying again @biergit 
We'll keep this issue open as gradle init run in the root of filesystem fails with a non actionnable error message.

**Comment by ljacomet** [comment]
This issue should have been marked as a duplicate of

* #15655

Note that it will be fixed in Gradle 8.1
